crates/pathfinder/src/sync/p2p.rs

```diff
@@ -61,10 +68,19 @@ impl Sync {
             .await
             .context("Persisting new Ethereum anchor")?;
 
+        let sync = self.clone();
+        tokio::spawn(async move {
+            // Sync missing transactions.
+            while let Err(e) = sync.sync_transactions().await {
+                tracing::info!(%e, "Error while syncing transactions, retrying");
+                tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+            }
+        });
+
```

Comment:
sync_headers below does not seem to restart upon error, I guess we should restart any sync future that fails